contracts/SwapKiwi.sol

```diff
@@ -200,43 +244,45 @@ contract SwapKiwi is Ownable, IERC721Receiver {
   }
 
   /**
-    * @dev Returns NFTs from SwapKiwi to swap initator.
-    *      Callable only if second user hasn't yet added NFTs.
-    *
-    * @param swapId ID of the swap that the swap participants want to cancel
-    */
-  function cancelSwap(uint256 swapId) external {
+  * @dev Returns NFTs from SwapKiwi to swap initator.
+  *      Callable only if second user hasn't yet added NFTs.
+  *
+  * @param swapId ID of the swap that the swap participants want to cancel
+  */
+  function cancelSwap(uint64 swapId) external {
     require(
```

Comment:
Swap needs to be read to memory in the beginning, and deleted from storage right away. Then function should continue working only with memory. In this way we can be absolutely sure that no reentrancy could affect things, and it will also make it cheaper.